src/js/view/view.js

```diff
@@ -945,30 +1002,50 @@ function View(_api, _model) {
         style(_wrapperElement, styles);
     }
 
-    this.stopFloating = function(forever) {
+    this.stopFloating = function(forever, mobileFloatIntoPlace) {
         if (forever) {
             _floatingConfig = null;
         }
         if (floatingPlayer === _playerElement) {
             floatingPlayer = null;
-
             _model.set('isFloating', false);
 
-            removeClass(_playerElement, 'jw-flag-floating');
-            onAspectRatioChange(_model, _model.get('aspectratio'));
-
-            // Wrapper should inherit from parent unless floating.
-            style(_playerElement, { backgroundImage: null }); // Reset to avoid flicker.
-            style(_wrapperElement, {
-                maxWidth: null,
-                width: null,
-                height: null,
-                left: null,
-                right: null,
-                top: null,
-                bottom: null,
-                margin: null
-            });
+            const resetFloatingStyles = () => {
+                removeClass(_playerElement, 'jw-flag-floating');
+                onAspectRatioChange(_model, _model.get('aspectratio'));
+
+                // Wrapper should inherit from parent unless floating.
+                style(_playerElement, { backgroundImage: null }); // Reset to avoid flicker.
+
+                style(_wrapperElement, {
+                    maxWidth: null,
+                    width: null,
+                    height: null,
+                    left: null,
+                    right: null,
+                    top: null,
+                    bottom: null,
+                    margin: null,
+                    transform: null,
+                    transition: null,
+                    'transition-timing-function': null
+                });
+            }
+
+            if (mobileFloatIntoPlace) {
+                // Reverses a dynamic animation where the top of the current player
+                // Smoothly transitions into the expected static space in the event
+                // we didn't start floating at 62px
+                style(_wrapperElement, {
+                    transform: `translateY(-${FLOATING_TOP_OFFSET - playerBounds.top}px)`,
+                    'transition-timing-function': 'ease-out'
+                });
+
+                setTimeout(resetFloatingStyles, 150);
```

Comment:
also know that there is hesitance around using `setTimeout` in general let alone for tracking the end of a floating animation however this approach allows us to dynamically ensure the player floats to the proper location.

given the fact that it's only `150ms` as well, there is little room for traditional concern around how the web threads lock up when exiting a tab or how it can get held up by other browser execution.

if anyone has an additional idea, more than happy to talk about it! I thought about using `requestAnimationFrame` but i would have to track deltas until we hit `150ms` which felt like overkill but i could be convinced otherwise

Reply:
Maybe use `transitionend` ? https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event

Reply:
:+1: great idea. works like a charm, but need a way of resetting styles for browsers that don't support that event, or could use 
```
webkitTransitionEnd otransitionend oTransitionEnd msTransitionEnd transitionend
```
but i'm not in love with that. let's chat monday!

Reply:
Don't forget `moz` 🦊 Maybe we can make a prefix util and also use it in css.js `transform()` and request-fullscreen-helper.js?